### PR TITLE
Add v2 terms confirmation flow

### PR DESCRIPTION
**Pull request by MelanijaGrunte**

### CODE REVIEW

File: app/src/main/java/lv/spkc/apturicovid/persistance/SharedPreferenceManager.kt [hunk 1]
```diff
@@ -19,7 +20,10 @@ class SharedPreferenceManager(private val preferenceStorage: PreferenceStorage)
 
     var isOnboardingFinished: Boolean
         get() = preferenceStorage.getObject(KEY_ONBOARDING, Boolean::class.java) ?: false
-        set(value) = preferenceStorage.setObject(KEY_ONBOARDING, value)
+        set(value) {
+            if (value) { preferenceStorage.setObject(KEY_ACCEPTANCE_V2_CONFIRMED, true) }
+            preferenceStorage.setObject(KEY_ONBOARDING, value)
```
Comment (edgarsMalahovskis) [inline app/src/main/java/lv/spkc/apturicovid/persistance/SharedPreferenceManager.kt:25]: acceptanceV2Confirmed = true

File: app/src/main/java/lv/spkc/apturicovid/ui/LanguageViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    val languageChangedEventLiveData: LiveData<Event<Boolean>> = _languageChangedEventLiveData
+
+    fun getSelectedLocale(): String {
+        return Locale.getDefault().language
+    }
+
+    fun setLanguage(language: String) {
+        sharedPreferenceManager.language = language
+        _languageChangedEventLiveData.value = Event(true)
+        _languageChangedLiveData.value = true
```
Comment (edgarsMalahovskis) [inline app/src/main/java/lv/spkc/apturicovid/ui/LanguageViewModel.kt:26]: Two live datas for the same task?

File: app/src/main/java/lv/spkc/apturicovid/ui/home/TermsConfirmationFragment.kt [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+            observeLiveData(viewModel.acceptancesLiveData) {
+                continueButton.isEnabled = it
+            }
+        }
+
+        viewLifecycleOwner.observeLiveData(languageViewModel.languageChangedLiveData) {
+            if (it) {
+                restartActivity()
+            }
```
Comment (edgarsMalahovskis) [inline app/src/main/java/lv/spkc/apturicovid/ui/home/TermsConfirmationFragment.kt:82]: if this live data onObserveLiveData will return true, we will restart activity.

File: app/src/main/java/lv/spkc/apturicovid/ui/home/TermsConfirmationFragment.kt [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+    private fun restartActivity() {
+        with(requireActivity()) {
+            val intent = Intent(this, MainActivity::class.java)
+            startActivity(intent)
+            finish()
+            overridePendingTransition(0, 0)
+        }
+    }
+}
```
Comment (edgarsMalahovskis) [inline app/src/main/java/lv/spkc/apturicovid/ui/home/TermsConfirmationFragment.kt:94]: this is not quite correct, first of all this restarts only mainActivity, secondly -> 
requireActivity().recreate()

File: app/src/main/java/lv/spkc/apturicovid/ui/intro/IntroFragment.kt [hunk 5]
```diff
@@ -41,95 +42,53 @@ class IntroFragment : BaseFragment() {
...
+                Pair(
+                    getString(R.string.privacy_link),
+                    object : DebouncedClickListener() {
+                        override fun performClick(v: View?) {
+                            val url =
+                                getString(R.string.privacy_policy_url) + languageViewModel.getSelectedLocale()
+                            openWebView(url)
+                        }
+                    })
+            )
```
Comment (edgarsMalahovskis) [inline app/src/main/java/lv/spkc/apturicovid/ui/intro/IntroFragment.kt:72]: Avoid WebView, use hyperlinks

File: app/src/main/res/drawable/image_world.xml [hunk 6]
```diff
@@ -0,0 +1,57 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (edgarsMalahovskis) [inline app/src/main/res/drawable/image_world.xml:1]: looks massive, maybe we should use image resource?

File: app/src/main/res/layout/fragment_data_transfer_submit.xml [hunk 7]
```diff
@@ -49,37 +48,52 @@
...
+                android:id="@+id/countries_list_btn"
+                android:layout_width="wrap_content"
+                android:layout_height="@dimen/default_button_height"
+                android:layout_marginStart="@dimen/fragment_side_margin"
+                android:layout_marginTop="8dp"
+                android:layout_marginEnd="@dimen/fragment_side_margin"
+                android:text="@string/label_list_of_countries"
+                android:textAlignment="viewStart"
+                android:textColor="@color/titleTextColor"
+                android:textSize="14sp"
```
Comment (edgarsMalahovskis) [inline app/src/main/res/layout/fragment_data_transfer_submit.xml:67]: dimen

File: app/src/main/res/layout/fragment_data_transfer_submit.xml [hunk 8]
```diff
@@ -49,37 +48,52 @@
...
-                app:layout_constraintTop_toBottomOf="@id/data_submit_description_tv" />
+                app:layout_constraintTop_toBottomOf="@id/countries_list_btn" />
 
             <com.alimuzaffar.lib.pin.PinEntryEditText
                 android:id="@+id/code_et"
-                android:layout_width="match_parent"
+                android:layout_width="0dp"
                 android:layout_height="60dp"
-                android:layout_marginStart="@dimen/fragment_side_margin"
+                android:layout_marginStart="25dp"
```
Comment (edgarsMalahovskis) [inline app/src/main/res/layout/fragment_data_transfer_submit.xml:94]: dimens

File: app/src/main/res/layout/fragment_data_transfer_submit.xml [hunk 9]
```diff
@@ -49,37 +48,52 @@
...
                 android:layout_marginTop="8dp"
-                android:layout_marginEnd="@dimen/fragment_side_margin"
+                android:layout_marginEnd="25dp"
```
Comment (edgarsMalahovskis) [inline app/src/main/res/layout/fragment_data_transfer_submit.xml:96]: dimens

File: app/src/main/res/layout/fragment_terms_confirmation.xml [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <ScrollView
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
```
Comment (edgarsMalahovskis) [inline app/src/main/res/layout/fragment_terms_confirmation.xml:11]: no no no, 0dp and constraints

**Review by edgarsMalahovskis — CHANGES_REQUESTED** [review]

**Review by edgarsMalahovskis — APPROVED** [review]

**Review by ULazdins — APPROVED** [review]

**Pull request merged**